modyn/selector/internal/selector_strategies/coreset_strategy.py

```diff
@@ -73,12 +91,20 @@ def _get_data(self) -> Iterable[list[int]]:
         assert isinstance(
             self._storage_backend, DatabaseStorageBackend
         ), "CoresetStrategy currently only supports DatabaseBackend"
+        # We set is_warmup here as well to support unit tests more easily.
+        # In end-to-end workflows, it should have been set by trigger() already.
+        if (self._next_trigger_id + 1) > self.warmup_triggers:
+            self.is_warmup = False
 
+        presampling_strategy = self.warmup_presampler if self.is_warmup else self.presampling_strategy
+        assert presampling_strategy is not None  # satisfy mypy
```

Comment:
we can make this `self.warmup_presampler` always a `NoPresamplingStrategy`, so that we can get rid of this assert.

We use `self.is_warmup` to decide wether to use `self.warmup_presampler` anyway, no need to make `self.warmup_presampler` optional.

Reply:
true, fixed.